src/migrate/Migrator.js

```diff
@@ -419,15 +440,42 @@ export default class Migrator {
   }
 
   setConfig(config) {
-    return assign({}, CONFIG_DEFAULT, this.config || {}, config);
+    const newConfig = assign({}, CONFIG_DEFAULT, this.config || {}, config);
+
+    // If migrationSource is not specified, but fs related config has been specified
+    // recreate the FsMigration source so the new filesystem related settings
+    // are used
+    if (
+      config &&
+      !config.migrationSource &&
+      (config.directory ||
+        config.sortDirsSeparately !== undefined ||
+        config.loadExtensions)
+    ) {
+      newConfig.migrationSource = new FsMigrations(
+        newConfig.directory,
```

Comment:
You are checking for values being set on config and then using values from newConfig. Even if this is still going to achieve same purpose, it reads confusingly. Also I still fail to see benefit of the check, considering that the outcome is always the same, we end up creating FsMigrations once with params from config.

Reply:
config is the values the user has passed in, newConfig is the merged config from existing, defaults and then user values.

The additional checks are ensuring the intention of the user works properly.